pwnlib/commandline/shellcraft.py

```diff
@@ -248,11 +187,11 @@ def main():
     code = pwnlib.asm.asm(code, arch = arch, os = os)
 
     if args.format in ['s', 'str', 'string']:
-        code = _string(code)
+        code = '"' + repr(code)[1:-1].replace('"', '\\"')  + '"\n'
```

Comment:
It still does not escape `$`.

The only point I can see of `--format string` is if you want it to be put in an `echo -e [foo]` or `echo $'[foo]'`. Right now that would not work.

But perhaps there are other usecases I did not forsee? :)